src/components/markdown/custom-components.tsx

```diff
@@ -54,11 +56,42 @@ export const SimpleLink = styled.a`
   text-decoration: none;
 
   &:hover {
-    color: ${theme.palette.text.link.hover};
     text-decoration: underline;
   }
 `;
 
+const AnchorIconStyled = styled(Anchor)`
+  display: inline-block;
+  position: absolute;
+  width: 0.5em;
+  height: 0.5em;
+  transform: translate(-0.75em, 0.5em);
+`;
+
+const AnchoredLink = (props) => {
+  return (
+    <SimpleLink {...props}>
+      <AnchorIconStyled />
+      {props.children}
+    </SimpleLink>
+  );
+};
+
+const AnchoredLinkStyled = styled(AnchoredLink)`
+  position: relative;
+  ${AnchorIconStyled} {
+     visibility: hidden;
+  }
+  
+  &:hover {
+    text-decoration: none;
+    ${AnchorIconStyled} {
+      visibility: visible;
+    }
+  }
+}
+`;
+
```

Comment:
this is a little bit more code then the previous inline svg solution but it fits pretty good into the existing way of having icon components and explicit components per use case.